src/main/java/com/marklogic/kafka/connect/source/XmlRowBatcherBuilder.java

```diff
@@ -0,0 +1,97 @@
+package com.marklogic.kafka.connect.source;
+
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.RowBatchSuccessListener;
+import com.marklogic.client.datamovement.RowBatcher;
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.client.io.DOMHandle;
+import com.marklogic.client.io.marker.ContentHandle;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Map;
+
+public class XmlRowBatcherBuilder extends LoggingObject implements RowBatcherBuilder<Document> {
+    private static final String TABLE_NS_URI = "http://marklogic.com/table";
+    private final String topic;
+    private final DataMovementManager dataMovementManager;
+    private final DocumentBuilderFactory documentBuilderFactory;
+
+    XmlRowBatcherBuilder(DataMovementManager dataMovementManager, Map<String, Object> parsedConfig) {
+        this.dataMovementManager = dataMovementManager;
+        this.topic = (String) parsedConfig.get(MarkLogicSourceConfig.TOPIC);
+        documentBuilderFactory = DocumentBuilderFactory.newInstance();
+    }
+
+    public RowBatcher<Document> newRowBatcher(List<SourceRecord> newSourceRecords) {
+        ContentHandle<Document> domHandle = new DOMHandle().withMimetype("application/xml");
+        RowBatcher<Document> rowBatcher =  dataMovementManager.newRowBatcher(domHandle);
+        rowBatcher.onSuccess(event -> onSuccessHandler(event, newSourceRecords));
+        return rowBatcher;
+    }
+
+    protected void onSuccessHandler(RowBatchSuccessListener.RowBatchResponseEvent<Document> event, List<SourceRecord> newSourceRecords) {
+        TransformerFactory transformerFactory = TransformerFactory.newInstance();
+        Transformer transformer;
+        try {
+            transformer = transformerFactory.newTransformer();
+        } catch (TransformerConfigurationException e) {
+            throw new RuntimeException("Unable to create XML Transformer: " + e.getMessage());
```

Comment:
Gotta catch and log this too. I think it's perfectly fine to just throw everything in here into one try/catch block, logging the generic error message I mentioned above. We really don't expect any errors here, so if there is an error, I think it's fine to just log it and keep going.